src/control/run_go_tests.sh

```diff
@@ -1,6 +1,61 @@
 #!/bin/bash
 ## Run linters across control plane code and execute Go tests
-set -e
+set -x
+
+function find_build_source()
+{
+	BASE=$PWD
+	
+	while true
+	do
+		path=`realpath ${BASE}`
+		if [ "${path}" == "/" ]; then
+			break
+		fi
+		test -e "${path}/.build_vars.sh" && echo "${path}/.build_vars.sh" && return
+		BASE=`dirname "${path}"`
+	done
+	echo ""
+}
+
+function check_environment()
+{
+	if [ -z "$LD_LIBRARY_PATH" ]; then
```

Comment:
Will there be an issue if this is set for other things?

Reply:
There might be an issue if you set this on your own with wrong values. However it should reset the values for the run with what it finds in .build_source.sh unless you've also set the two cgo environment variables.

Reply:
> Will this work with RPMs?

It might not but I don't think it currently works with the RPMs either. All of our test running scripts require that you have a full source tree since it looks for a build artifact to determine the locations of all of these things. There is no conditional check in run_tests.sh to see if anything is set before just sourcing ./.build_sources.sh